lib/semian/lru_hash.rb

```diff
@@ -83,16 +89,36 @@ def clear_unused_resources
     return unless @lock.try_lock
     # Clears resources that have not been used in the last 5 minutes.
     begin
+      timer_start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+
+      payload = {
+          size: @table.size,
+          examined: 0,
+          cleared: 0,
+          elapsed: nil,
+      }
+
+      stop_time = Time.now - MINIMUM_TIME_IN_LRU # Don't process resources updated after this time
       @table.each do |_, resource|
-        break if resource.updated_at + MINIMUM_TIME_IN_LRU > Time.now
+        payload[:examined] += 1
+
+        # The update times of the resources in the LRU are monotonically increasing,
+        # time, so we can stop looking once we find the first resource with an
+        # update time after the stop_time.
+        break if resource.updated_at > stop_time
+
+        # TODO(michaelkipper): Should this be a flag?
         next if resource.in_use?
 
         resource = @table.delete(resource.name)
         if resource
-          Semian.notify(:lru_hash_cleaned, self, :cleaning, :lru_hash)
+          payload[:cleared] += 1
           resource.destroy
         end
       end
+
+      payload[:elapsed] = Process.clock_gettime(Process::CLOCK_MONOTONIC) - timer_start
+      Semian.notify(:lru_hash_gc, self, nil, nil, payload)
```

Comment:
This event breaks backwards compatibility (as you see in `grpc_test`). The reason is that it's only temporary?

Reply:
I think we'd have to pass a lot of state around to get around that.
We could optionally pass something other than `nil` for `scope` and `adapter`, if you thought that was necessary, but it would still break those tests.